site/app/controllers/PollController.php

```diff
@@ -503,6 +505,88 @@ public function openPoll(): RedirectResponse {
+        }
+        elseif ($poll->getAllowsCustomResponses() === false) {
+            return JsonResponse::getFailResponse("Poll is currently not accepting custom responses");
+        }
+        elseif ($option_repo->existsByPollAndResponse($poll_id, $poll_response)) {
+            return JsonResponse::getFailResponse("A similar response already exists");
+        }
+
+        $custom_poll_option = new Option($poll->getOptions()->count(), $poll_response, $poll->isSurvey(), $user_id);
+        $poll->addOption($custom_poll_option);
+        $em->persist($custom_poll_option);
+
+        $response = new Response($user_id);
+        $poll->addResponse($response, $custom_poll_option->getId());
+        $em->persist($response);
+        $em->flush();
+
+        return JsonResponse::getSuccessResponse(["message" => "Successfully added custom response"]);
+    }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/polls/removeCustomResponse", methods={"POST"})
+     */
+    public function removeCustomResponse(): JsonResponse {
+        $poll_id = intval($_POST['poll_id'] ?? -1);
+        $option_id = intval($_POST['option_id'] ?? -1);
+        $user_id = $this->core->getUser()->getId();
+        $em = $this->core->getCourseEntityManager();
+        $poll_repo = $em->getRepository(Poll::class);
+        /** @var Poll|null */
+        $poll = $poll_repo->find($poll_id);
+        if ($poll === null) {
+            return JsonResponse::getErrorResponse("Invalid Poll ID");
+        }
+        elseif (!$poll->isOpen() && !$this->core->getUser()->accessFaculty()) {
+            return JsonResponse::getFailResponse("Poll is closed");
+        }
+
+        /** @var Option|null */
+        $custom_option = $this->core->getCourseEntityManager()->find(Option::class, $option_id);
+        if ($custom_option === null) {
+            return JsonResponse::getErrorResponse("Could not find custom response");
+        }
+        elseif ($custom_option->getAuthorId() !== $user_id && !$this->core->getUser()->accessFaculty()) {
+            return JsonResponse::getErrorResponse("You have no access to remove this custom response");
+        }
+        elseif ($custom_option->getUserResponses()->count() > 1 || ($custom_option->getUserResponses()->count() === 1 && $custom_option->getUserResponses()->first()->getStudentId() !== $user_id)) {
+            return JsonResponse::getErrorResponse("Cannot delete response option that has already been submitted as an answer by another individual");
+        }
+
+        foreach ($custom_option->getUserResponses() as $response) {
+            $em->remove($response);
+        }
+        $poll->removeOption($custom_option);
+        $em->remove($custom_option);
+        $em->persist($poll);
+        $em->flush();
+
+        return JsonResponse::getSuccessResponse(["message" => "Successfully removed custom response"]);
+    }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/polls/setEnded", methods={"POST"})
```

Comment:
This can be removed since it has been changed to a php8 route

Reply:
[a6735ec](https://github.com/Submitty/Submitty/pull/10197/commits/a6735ec8ff21da1dd55256759150bc150e2fd47f)